### ISSUE TITLE
[14.0] nl tax statement - Error since upgrade from 13 to 14

### ISSUE THREAD

**Issue by E4eqRL0yw0yajQJ433QL** [OCA/l10n-netherlands#350]
## Module
l10n_nl_vat_statement.py

## Describe the bug
System creates error when running the Dutch btw report functionality
  File "/var/lib/odoo/addons/14.0/l10n_nl_tax_statement/models/l10n_nl_vat_statement.py", line 410, in statement_update
    self._set_statement_lines(lines, move_lines)
  File "/var/lib/odoo/addons/14.0/l10n_nl_tax_statement/models/l10n_nl_vat_statement.py", line 437, in _set_statement_lines
    tags_map = self._get_tags_map()
  File "/var/lib/odoo/addons/14.0/l10n_nl_tax_statement/models/l10n_nl_vat_statement.py", line 388, in _get_tags_map
    matching.update({tag.id: (res_code.group(0), "omzet")})

## To Reproduce
Affected version: 14.0 (earlier this problem was already resolved in version 13.0)

1. Difficult to reproduce, since in a vanilla environment this issue does not occur
2.  I had this problem before (after upgrade from 12 -> 13 and at that time it was fixed by making the code change as described below.
3. Problem came up directly after migrating: I simply ran the Dutch btw report and clicked "refresh" to find the relevant vat lines. That led to the error as described above.
4. This happened both in the existing declarations and in a new declaration.

**Expected behavior**
I had expected that the system would have found the corresponding btw line items

**Proposed code change (based on the same problem in version 13.0) - I tested it and it resolves the problem in 14.0.
Module: l10n_nl_vat_statement.py
Start of removal
        for tag in nl_tags:
            res_code = pattern_code.match(tag.name)
            if re.search("omzet", tag.name, re.IGNORECASE):
                matching.update({tag.id: (res_code.group(0), "omzet")})
            elif re.search("btw", tag.name, re.IGNORECASE):
                matching.update({tag.id: (res_code.group(0), "btw")})
            elif res_code:
                matching.update({tag.id: (res_code.group(0), False)})
End of removal

Start of insertion
        for tag in nl_tags:
            res_code = pattern_code.match(tag.name)
            if res_code and re.search("omzet", tag.name, re.IGNORECASE):
                matching.update({tag.id: (res_code.group(0), "omzet")})
            elif res_code and re.search("btw", tag.name, re.IGNORECASE):
                matching.update({tag.id: (res_code.group(0), "btw")})
            elif res_code:
                matching.update({tag.id: (res_code.group(0), False)})
End of insertion

**Additional context**
See [https://github.com/OCA/l10n-netherlands/issues/318] for the same behaviour and corresponding fix in version 13.

**Comment by astirpe** [comment]
The error mentioned in the description of the bug is caused by wrong account tags for taxes, as explained also in 
https://github.com/OCA/l10n-netherlands/issues/318 so that error is a "symptom" that something is wrong.

The proposed code change is just hiding the error but is not fixing the root of the problem which is the wrong data (tags for taxes). We need a proper fix of the issue, that would be the fixing of the data, like Cas was doing, with a dedicated script. But for that we need to analyze what's going wrong in that particular database.

The proposed code change is still valid IMO and I'm thinking of including it when porting the module to V15, however that could cause a more difficult detection of wrong tax tags.

**Comment by E4eqRL0yw0yajQJ433QL** [comment]
Thanks for your kind comments Andrea. I did manually make the code change in V14, so for me it is now working as desired. I THOUGHT I fixed all the account tag problems already in V13, but from your comment I derive that there still might be issues there. I'd be happy to provide access to my database if that would benefit analyzing the real root-cause of the problem. Just let me know what could be convenient for you.
